Src/AwesomeAssertions/Primitives/StringEndStrategy.cs

```diff
@@ -19,24 +20,70 @@ public StringEndStrategy(IEqualityComparer<string> comparer, string predicateDes
 
     public void ValidateAgainstMismatch(AssertionChain assertionChain, string subject, string expected)
     {
-        assertionChain
-            .ForCondition(subject!.Length >= expected.Length)
-            .FailWith($"{ExpectationDescription}{{0}}{{reason}}, but {{1}} is too short.", expected, subject);
-
-        if (!assertionChain.Succeeded)
+        var (mismatchInSubject, mismatchInExpectation) = IndexOfLastMismatch(subject, expected, comparer);
+        if (mismatchInExpectation < 0)
         {
             return;
         }
 
-        int indexOfMismatch = subject.Substring(subject.Length - expected.Length).IndexOfFirstMismatch(expected, comparer);
+        var (target, targetIndex) = subject.Length >= expected.Length
```

Comment:
```suggestion
        var mismatchLocationDescription = subject.Length >= expected.Length
            ? $"before index {mismatchInSubject} of actual"
            : $"before index {mismatchInExpectation} of expected";
```
This `mismatchLocationDescription` is intended to be used below in the settings initializer.

I think, you tried to over-optimize the redundancies. But, here it is too much.
I tried to build this suggestion, unsure whether it is correct.
For me, it is more readable at least. But, still I do not understand, why the result depends on the string lengths. I', quite sure that it is correct. But, I hope some words of comment should help us to understand the logic in future. For you, currently working heavily on this feature, it is quite easy.

Reply:
For me adding `of actual` or `of expected` to the description partially defeats the aim of making the diff easier to understand.

For me, we must have
```csharp
string mismatchDescription = subject.Length >= expected.Length
    ? $"they differ before index {mismatchInSubject}"
    : $"actual was too short";
```
`index` must **always** be refer to the subject, not "it depends on".

That's currently my only "must change" left.

Reply:
I agree with the overall sentiments here that this isn't really adding much clarity, I'm not sure if still we need to change the description if the subject is too short in this case because we already have the subject index. For example:

```
                """
                *before index 0*
                           ↓ (actual)
                          "H"
                  "ABCDEFGHI"
                           ↑ (expected).
                """
```

Reply:
I wrote down the two cases for this check, both for `StartWith` and `EndWith`:
1. Subject fully matches expected, but is shorter
2. Subject partially matches expected, and is shorter

When proposing the "too short" message, I had only thought about 1, not two.

I think, if we always refer to the subject index, we should be fine:

1. The whole subject matches part of the expected, but is too short:
```
"H".Should().EndWith("ABCDEFGH")

*before index 0*
         ↓ (actual)
         "H"
  "ABCDEFGH"
         ↑ (expected).
```
2. The subject only matches parts of the expected, and is too short:
```
"FooH".Should().EndWith("ABCDEFGH")

*before index 3*
         ↓ (actual)
      "FooH"
  "ABCDEFGH"
         ↑ (expected).
```

Same considerations for `StartWith`:

1. The whole subject matches part of the expected, but is too short:

```
"A".Should().StartWith("ABCDEFGH")

*at index 1*
    ↓ (actual)
  "A"
  "ABCDEFGH"
    ↑ (expected).
```

2. The subject only matches parts of the expected, and is too short:

```
"AFoo".Should().StartWith("ABCDEFGH")

*at index 1*
    ↓ (actual)
  "AFoo"
  "ABCDEFGH"
    ↑ (expected).
```